### PR TITLE
Add --custom-log-dict CLI option

### PR DESCRIPTION
**Pull request by isobelhooper**
This points at an import path for a dictionary (e.g. myproject.logging.CONFIG_DICT) that Arq can configure its logging with instead of using the default configuration.

This would be useful for e.g. configuring Arq logging output to use [python-json-logger](https://github.com/madzak/python-json-logger).

We use this in a fork of Arq for a project I'm working on, and I wanted to contribute this back upstream. I'm happy to make any changes that are needed.

### CODE REVIEW

File: arq/cli.py [hunk 1]
```diff
@@ -28,15 +30,24 @@
...
     """
     sys.path.append(os.getcwd())
     worker_settings_ = cast('WorkerSettingsType', import_string(worker_settings))
-    logging.config.dictConfig(default_log_config(verbose))
+    if custom_log_dict:
+        try:
+            config_path, config_dict = custom_log_dict.rsplit('.', maxsplit=1)
+            log_config = getattr(importlib.import_module(config_path), config_dict)
+        except (TypeError, AttributeError):
+            log_config = default_log_config(verbose)
```
Comment (samuelcolvin) [inline arq/cli.py:47]: we should log (probably just print to sterr is easiest) a warning in this case with details of what went wrong.

Or we could just raise an error.

**Review by samuelcolvin — COMMENTED** [review]
I'm okay with this, needs also:
* a test
* to be rebased to master - should fix build

File: arq/cli.py (hunk 1 above)
Comment (isobelhooper) [reply arq/cli.py:47]: Raising an ImportError from `import_string` seems like it would work okay - I'll do that.

**Comment by samuelcolvin** [comment]
you've got an error in CI on py3.10. You (or I) will also need to pin jinja to make docs build.

**Comment by samuelcolvin** [comment]
update, I've pinned jinja in master.

**Comment by isobelhooper** [comment]
This is now rebased on current master, and I've also updated the version of black to deal with an incompatibility with certain Click versions. Let me know if there's anything else you'd like me to change in this.

**Comment by samuelcolvin** [comment]
thanks so much.

**Pull request merged**